File: _scratch/run/score_of_parentheses/sol.js
/**
 * @param {string} s
 * @return {number}
 */
var scoreOfParentheses = function(s) {
    const st = [0];
    for (const c of s) {
        if (c === '(') st.push(0);
        else {
            const v = st.pop();
            st[st.length - 1] += Math.max(2 * v, 1);
        }
    }
    return st[0];
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(String(scoreOfParentheses(_s)));}
console.log(_o.join('\n'));

File: _scratch/run/score_of_parentheses/sol.py
class Solution:
    def scoreOfParentheses(self, s):
        st = [0]
        for c in s:
            if c == "(":
                st.append(0)
            else:
                v = st.pop()
                st[-1] += max(2 * v, 1)
        return st[0]
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append(str(Solution().scoreOfParentheses(_s)))
print('\n'.join(_o))